pkg/daemon/controller/controller.go

```diff
@@ -265,6 +271,75 @@ func (sc *ServiceController) Reconcile(_ context.Context, req ctrl.Request) (res
 	return ctrl.Result{}, nil
 }
 
+// reconcileEnvironmentVariables makes sure that the proxy variables exist as expected, or are safely rectified
+func (sc *ServiceController) reconcileEnvironmentVariables(envVars map[string]string, node core.Node) error {
+	restartRequired, err := sc.ensureVarsAreUpToDate(envVars)
+	if err != nil {
+		return err
+	}
+	if restartRequired {
+		// Applying the reboot annotation results in an event picked up by WMCO's node controller to reboot the instance
+		if err = metadata.ApplyRebootAnnotation(sc.ctx, sc.client, node); err != nil {
+			return fmt.Errorf("error setting reboot annotation on node %s: %w", sc.nodeName, err)
+		}
+	}
+	// Wait until the environment variables at the process level are set as expected.
+	// This will only be picked up after WMCO reboots the instance
+	err = wait.PollUntilContextTimeout(sc.ctx, 15*time.Second, 2*time.Minute, true,
+		func(ctx context.Context) (done bool, err error) {
+			stillNeedsReboot := false
+			for _, varName := range cluster.SupportedProxyVars {
+				cmd := fmt.Sprintf("[Environment]::GetEnvironmentVariable('%s', 'Process')", varName)
+				out, err := sc.psCmdRunner.Run(cmd)
+				if err != nil {
+					return false, fmt.Errorf("error running PowerShell command %s with output %s: %w", cmd, out, err)
+				}
+				if strings.TrimSpace(out) != envVars[varName] {
+					stillNeedsReboot = true
+				}
+			}
+			return !stillNeedsReboot, nil
+		})
+	if err != nil {
+		return fmt.Errorf("error waiting for environment vars to get picked up by processes: %w", err)
```

Comment:
All processes forked from the default shell. For clarity the poll above checks the env vars in the shell process. Since child processes inherit from the parent processes, and all the processes we care about (containerd, kubelet, etc) are all spawned from the shell, this is a sufficient check